velero-csi-plugin/csi_restorer.go

```diff
@@ -63,16 +66,35 @@ func (p *CSIRestorer) Execute(input *velero.RestoreItemActionExecuteInput) (*vel
 		return nil, errors.Errorf("Could not find volume snapshot name on PVC")
 	}
 
-	g := "snapshot.storage.k8s.io"
-	pvc.Spec.DataSource = &corev1api.TypedLocalObjectReference{
-		// This needs to be a pointer, since nil is used for the default group
-		APIGroup: &g,
-		Kind:     "VolumeSnapshot",
-		Name:     volumeSnapshotName,
+	for k := range annotations {
```

Comment:
I'm not sure that we need L54-L62 -- can't we just iterate over `pvc.Annotations` since we've converted to a structured type? The meta accessor seems superfluous now.